psi4/driver/qcdb/intf_dftd3/dashparam.py

```diff
@@ -0,0 +1,598 @@
+        level_hint = level_hint.lower()
+        try:
+            dashlevel_candidate_1 = get_dispersion_aliases()[level_hint]
+        except KeyError:
+            raise ValidationError("""Requested -D correction level ({}) not among ({})""".format(
+                level_hint, dashcoeff.keys()))
+
+    if name_hint is None:
+        dashlevel_candidate_2 = None
+        name_key = None
+        disp_params = {}
+    else:
+        name_hint = name_hint.lower()
+        trial_split = name_hint.rsplit('-', 1)
+
+        if name_hint in get_dispersion_aliases():
+            # case disp-only: d3, d3zero, d3(bj)
+            dashlevel_candidate_2 = get_dispersion_aliases()[name_hint]
+            name_key = None
+            disp_params = {}
+        elif (dashcoeff_supplement is not None) and name_hint in supplement_dashlevel_lookup:
+            # case fctldisp: wb97x-d, hf+d
+            dashlevel_candidate_2 = supplement_dashlevel_lookup[name_hint]
+            name_key = name_hint
+            disp_params = dashcoeff_supplement[dashlevel_candidate_2]['definitions'][name_hint]
+        elif (len(trial_split) == 2 and trial_split[1] in get_dispersion_aliases()
+              and trial_split[0] in dashcoeff[get_dispersion_aliases()[trial_split[1]]]['definitions']):
+            # case fctldisp: b3lyp-d3, b3lyp-d3zero, b3lyp-d3(bj)
+            dashlevel_candidate_2 = get_dispersion_aliases()[trial_split[1]]
+            name_key = trial_split[0]
+            disp_params = dashcoeff[dashlevel_candidate_2]['definitions'][trial_split[0]]
+        elif (level_hint is not None) and name_hint in dashcoeff[dashlevel_candidate_1]['definitions']:
+            # case fctl: b3lyp
+            dashlevel_candidate_2 = None
+            name_key = name_hint
+            disp_params = dashcoeff[dashlevel_candidate_1]['definitions'][name_hint]
+        elif name_hint == '':
+            dashlevel_candidate_2 = None
+            name_key = None
+            disp_params = {}
+        else:
+            dashlevel_candidate_2 = None
+            raise ValidationError("""Can't guess -D correction level from ({})""".format(name_hint))
+
+    disp_params = copy.deepcopy(disp_params)
+    # TODO citation = disp_params.pop('citation', None)
+
+    if dashlevel_candidate_1 is None and dashlevel_candidate_2 is None:
+        raise ValidationError(
+            """Can't guess -D correction level from name_hint ({}) and level_hint ({})""".format(level_hint))
+    elif dashlevel_candidate_1 is not None and dashlevel_candidate_2 is not None:
+        if dashlevel_candidate_1 != dashlevel_candidate_2:
+            raise ValidationError(
+                """Inconsistent -D correction level from name_hint ({}) and level_hint ({})""".format(
+                    dashlevel_candidate_2, dashlevel_candidate_1))
+    dashleveleff = dashlevel_candidate_1 or dashlevel_candidate_2
+
+    try:
+        allowed_params = dashcoeff[dashleveleff]['default'].keys()
+    except KeyError:
```

Comment:
I wonder when would this ever be the case. Aren't both `dashlevel_candidate_{1,2}` populated from `get_dispersion_aliases()`?

Reply:
I think you're right. May have been a remanent of earlier logic (been a few passes). I'll remove it, and at worst, it'll `KeyError`, not proceed on a wrong path.